ee/clickhouse/queries/retention/retention_event_query.py

```diff
@@ -74,7 +94,8 @@ def get_query(self) -> Tuple[str, Dict[str, Any]]:
             {f"AND {entity_query}"}
             {f"AND {date_query}" if self._event_query_type != RetentionQueryType.TARGET_FIRST_TIME else ''}
             {prop_query}
-            {f"GROUP BY target HAVING {date_query}" if self._event_query_type == RetentionQueryType.TARGET_FIRST_TIME else ''}
+            ORDER BY event_date, target
+            {f"LIMIT 1 BY target" if self._event_query_type == RetentionQueryType.TARGET_FIRST_TIME else ''}
```

Comment:
Is the `LIMIT 1 BY ` change purely for performance or is the result different?

Reply:
Yeah I'll add a benchmark label, I might just revert the limit by change 🤔